apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/faucet/components/SpinWheel.tsx

```diff
@@ -0,0 +1,191 @@
+
+  if (seconds > 60 && seconds < 3600) {
+    return `${minutes}m`;
+  }
+
+  return `${hours}h ${minutes}m`;
+}
+
+export function TestnetSpinWheel({
+  chain,
+  ttlSeconds,
+}: {
+  chain: ChainMetadataWithServices;
+  ttlSeconds: number;
+}) {
+  const address = useActiveAccount()?.address;
+  const chainId = chain.chainId;
+  const definedChain = defineChain(chainId);
+  const { data: evmBalance, isLoading } = useWalletBalance({
+    address: THIRDWEB_ENGINE_FAUCET_WALLET,
+    chain: definedChain,
+    client: thirdwebClient,
+  });
+  const router = useRouter();
+
+  const claimMutation = useMutation({
+    mutationFn: async (result: string) => {
+      try {
+        const amount = result === "Come back tomorrow" ? "0" : result;
+        const requestBody = {
+          chainId: chainId,
+          toAddress: address,
+          amount: amount,
+        };
+
+        const response = await fetch("/api/testnet-faucet/claim", {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify(requestBody),
+        });
+
+        if (!response.ok) {
+          const errorText = await response.json();
+          toast.error(errorText.error);
+          return;
+        }
+
+        router.refresh();
+        const successMessage = `${amount} ${chain?.nativeCurrency?.name} sent to ${address}`;
+        toast.success(successMessage);
+      } catch (error) {
+        toast.error("An unexpected error occurred.");
+        throw error;
+      }
+    },
+  });
+
+  const isFaucetEmpty =
+    evmBalance?.value !== undefined && evmBalance.value < 100000000000000000n;
+
+  const faucetMessage = isFaucetEmpty ? (
+    <div>
+      <p>
+        There is no {chain.nativeCurrency.symbol} available. Please send funds
+        to the following address:
+      </p>
+      <div className="h-2" />
+      <CopyTextButton
+        textToCopy={THIRDWEB_ENGINE_FAUCET_WALLET}
+        textToShow={THIRDWEB_ENGINE_FAUCET_WALLET}
+        tooltip="Copy"
+        className="text-lg -translate-x-2 py-0.5"
+        copyIconPosition="right"
+      />
+    </div>
+  ) : (
+    <p>You may claim {chain.nativeCurrency.symbol} once every 24 hours.</p>
+  );
+
+  const claimButtonCta =
+    ttlSeconds > 0
+      ? `Your next claim is available in ${formatTime(ttlSeconds)}`
+      : `Spin for ${chain.nativeCurrency.symbol}`;
+  const closeButtonCta = ttlSeconds > 0 ? "Done" : "Close";
+
+  const spinWheelProps: ISpinWheelProps = {
+    segments,
+    onFinished: claimMutation.mutateAsync,
+    primaryColor: "black",
+    contrastColor: "white",
+    buttonText: "💜",
+    isOnlyOnce: false,
+    size: 290,
+    // Used to randomize result.
+    upDuration: Math.random() * 100 + 100,
+    downDuration: 600,
+    arrowLocation: "top",
+    showTextOnSpin: false,
+    isSpinSound: false,
+  };
+
+  if (!address) return <CustomConnectWallet />;
+
+  return (
+    <>
+      {isLoading ?? <Spinner />}
```

Comment:
was expecting this to be `isLoading ? <spinner> : <content if not loading>` why is it different than that?

Reply:
For some reason when I do that the spinner takes over the screen

https://github.com/user-attachments/assets/57c8ff4e-1381-4ff5-9d17-94ecc90b6a80